drivers/gpio/gpio-pwm.c

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * GPIO driver wrapping PWM API
+ *
+ * PWM 0% and PWM 100% are equivalent to digital GPIO
+ * outputs, and there are times where it is useful to use
+ * PWM outputs as straight GPIOs (eg outputs of NXP PCA9685
+ * I2C PWM chip). This driver wraps the PWM API as a GPIO
+ * controller.
+ *
+ * Copyright (C) 2021 Raspberry Pi (Trading) Ltd.
+ */
+
+#include <linux/err.h>
+#include <linux/gpio/driver.h>
+#include <linux/module.h>
+#include <linux/platform_device.h>
+#include <linux/pwm.h>
+
+#define NUM_GPIO 1
```

Comment:
As per the first comment, I'd love to get rid of the define and read it from the array size of ```pwms```, I just couldn't find the right API calls to do so.

Ideally I want a ```devm_pwm_get``` call that takes an index, but it looks like the PWM API doesn't allow that.
https://elixir.bootlin.com/linux/latest/source/drivers/pwm/core.c#L756
```
 * of_pwm_get() - request a PWM via the PWM framework
 * @dev: device for PWM consumer
 * @np: device node to get the PWM from
 * @con_id: consumer name
 *
 * Returns the PWM device parsed from the phandle and index specified in the
 * "pwms" property of a device tree node or a negative error-code on failure.
 * Values parsed from the device tree are stored in the returned PWM device
 * object.
 *
 * If con_id is NULL, the first PWM device listed in the "pwms" property will
 * be requested. Otherwise the "pwm-names" property is used to do a reverse
 * lookup of the PWM index. This also means that the "pwm-names" property
 * becomes mandatory for devices that look up the PWM device via the con_id
 * parameter.
```
So it's pwm-names, or only one pwm :-(
I can see an array of names "gpio0", "gpio1", "gpio2" in my future.

Reply:
It does sound a bit painful, but with `of_property_count_strings` and `of_property_read_string_index` at least your driver won't need to hard-code any names - just the overlay.